presto-hive/src/main/java/com/facebook/presto/hive/util/InternalHiveSplitFactory.java

```diff
@@ -160,14 +159,23 @@ public String getPartitionName()
                 // skip zero-width block, except in the special circumstance: slice is empty, and the block covers the empty slice interval.
                 continue;
             }
-            blockBuilder.add(new InternalHiveBlock(blockEnd, getHostAddresses(blockLocation)));
+
+            List<HostAddress> addresses = getHostAddresses(blockLocation);
+            if (!schedulerUsesHostAddresses && !(forceLocalScheduling && hasRealAddress(addresses))) {
```

Comment:
I'm wondering if this predicate could be extracted to a method named `needsHostAddresses`.